dotnet/src/AutoGen.Mistral/Agent/MistralClientAgent.cs

```diff
@@ -0,0 +1,106 @@
+    private ToolChoiceEnum? _toolChoice;
+    public MistralClientAgent(
+        MistralClient client,
+        string name,
+        string model,
+        string systemMessage = "You are a helpful AI assistant",
+        int? randomSeed = null,
+        ToolChoiceEnum? toolChoice = null,
+        bool jsonOutput = false)
+    {
+        _client = client;
+        Name = name;
+        _systemMessage = systemMessage;
+        _model = model;
+        _randomSeed = randomSeed;
+        _jsonOutput = jsonOutput;
+        _toolChoice = toolChoice;
+    }
+
+    public string Name { get; }
+
+    public async Task<IMessage> GenerateReplyAsync(
+        IEnumerable<IMessage> messages,
+        GenerateReplyOptions? options = null,
+        CancellationToken cancellationToken = default)
+    {
+        var request = BuildChatRequest(messages, options);
+        var response = await _client.CreateChatCompletionsAsync(request);
+
+        return new MessageEnvelope<ChatCompletionResponse>(response, from: this.Name);
+    }
+
+    public async Task<IAsyncEnumerable<IStreamingMessage>> GenerateStreamingReplyAsync(
+        IEnumerable<IMessage> messages,
+        GenerateReplyOptions? options = null,
+        CancellationToken cancellationToken = default)
+    {
+        var request = BuildChatRequest(messages, options);
+        var response = _client.StreamingChatCompletionsAsync(request);
+
+        return ProcessMessage(response);
+    }
+
+    private async IAsyncEnumerable<IMessage> ProcessMessage(IAsyncEnumerable<ChatCompletionResponse> response)
+    {
+        await foreach (var content in response)
+        {
+            yield return new MessageEnvelope<ChatCompletionResponse>(content, from: this.Name);
+        }
+    }
+
+    private ChatCompletionRequest BuildChatRequest(IEnumerable<IMessage> messages, GenerateReplyOptions? options)
+    {
+        var chatHistory = BuildChatHistory(messages);
+        var chatRequest = new ChatCompletionRequest(model: _model, messages: chatHistory.ToList(), temperature: options?.Temperature, randomSeed: _randomSeed)
+        {
+            MaxTokens = options?.MaxToken,
+            ResponseFormat = _jsonOutput ? new ResponseFormat() { ResponseFormatType = "json_object" } : null,
+        };
+
+        if (options?.Functions != null)
+        {
+            chatRequest.Tools = options.Functions.Select(f => new FunctionTool(f.ToMistralFunctionDefinition())).ToList();
+            chatRequest.ToolChoice = _toolChoice ?? ToolChoiceEnum.Auto;
+        }
+
+        return chatRequest;
+    }
+
+    private IEnumerable<ChatMessage> BuildChatHistory(IEnumerable<IMessage> messages)
+    {
+        var history = messages.Select(m => m switch
+        {
+            IMessage<ChatMessage> chatMessage => chatMessage.Content,
+            _ => throw new ArgumentException("Invalid message type")
+        });
+
+        // if there's no system message in the history, add one to the beginning
```

Comment:
Is this a little unexpected for users if both _systemMessage and message in the chat history is set? I have encountered scenarios where task-specific system message instructions are needed. e.g., in a sequence of different chats with the same agent, each chat has different instructions.